common/src/main/cpp/classes/openxr_fb_spatial_entity.cpp

```diff
@@ -0,0 +1,340 @@
+	(*userdata)->emit_signal("set_component_enabled_completed", XR_SUCCEEDED(p_result), from_openxr_component_type(p_component), p_enabled);
+	memdelete(userdata);
+}
+
+PackedStringArray OpenXRFbSpatialEntity::get_semantic_labels() const {
+	return OpenXRFbSceneExtensionWrapper::get_singleton()->get_semantic_labels(space);
+}
+
+Dictionary OpenXRFbSpatialEntity::get_room_layout() const {
+	OpenXRFbSceneExtensionWrapper::RoomLayout room_layout;
+	if (!OpenXRFbSceneExtensionWrapper::get_singleton()->get_room_layout(space, room_layout)) {
+		return Dictionary();
+	}
+
+	Dictionary ret;
+	ret["floor"] = OpenXRUtilities::uuid_to_string_name(room_layout.floor);
+	ret["ceiling"] = OpenXRUtilities::uuid_to_string_name(room_layout.ceiling);
+
+	Array walls_array;
+	walls_array.resize(room_layout.walls.size());
+	for (int i = 0; i < room_layout.walls.size(); i++) {
+		walls_array[i] = OpenXRUtilities::uuid_to_string_name(room_layout.walls[i]);
+	}
+	ret["walls"] = walls_array;
+
+	return ret;
+}
+
+Array OpenXRFbSpatialEntity::get_contained_uuids() const {
+	Vector<XrUuidEXT> uuids = OpenXRFbSpatialEntityContainerExtensionWrapper::get_singleton()->get_contained_uuids(space);
+
+	Array ret;
+	ret.resize(uuids.size());
+	for (int i = 0; i < uuids.size(); i++) {
+		ret[i] = OpenXRUtilities::uuid_to_string_name(uuids[i]);
+	}
+	return ret;
+}
+
+Rect2 OpenXRFbSpatialEntity::get_bounding_box_2d() const {
+	return OpenXRFbSceneExtensionWrapper::get_singleton()->get_bounding_box_2d(space);
+}
+
+AABB OpenXRFbSpatialEntity::get_bounding_box_3d() const {
+	return OpenXRFbSceneExtensionWrapper::get_singleton()->get_bounding_box_3d(space);
+}
+
+PackedVector2Array OpenXRFbSpatialEntity::get_boundary_2d() const {
+	return OpenXRFbSceneExtensionWrapper::get_singleton()->get_boundary_2d(space);
+}
+
+void OpenXRFbSpatialEntity::track() {
+	ERR_FAIL_COND_MSG(!is_component_enabled(COMPONENT_TYPE_LOCATABLE), vformat("Cannot track spatial entity %s because COMPONENT_TYPE_LOCATABLE isn't enabled.", uuid));
+	OpenXRFbSpatialEntityExtensionWrapper::get_singleton()->track_entity(uuid, space);
+}
+
+void OpenXRFbSpatialEntity::untrack() {
+	OpenXRFbSpatialEntityExtensionWrapper::get_singleton()->untrack_entity(uuid);
+}
+
+bool OpenXRFbSpatialEntity::is_tracked() const {
+	return OpenXRFbSpatialEntityExtensionWrapper::get_singleton()->is_entity_tracked(uuid);
+}
+
+MeshInstance3D *OpenXRFbSpatialEntity::create_mesh_instance() const {
+	MeshInstance3D *mesh_instance = nullptr;
+
+	if (is_component_enabled(COMPONENT_TYPE_BOUNDED_3D)) {
+		Ref<BoxMesh> box_mesh;
+		box_mesh.instantiate();
+
+		AABB bounding_box = get_bounding_box_3d();
+		box_mesh->set_size(bounding_box.size);
+
+		mesh_instance = memnew(MeshInstance3D);
+		mesh_instance->set_mesh(box_mesh);
+		mesh_instance->set_position(bounding_box.get_center());
+
+	} else if (is_component_enabled(COMPONENT_TYPE_BOUNDED_2D)) {
+		Ref<PlaneMesh> plane_mesh;
```

Comment:
Out of curiosity, wouldn't a quad mesh be more appropriate here?

Reply:
Yeah `Quadmesh` is a subclass of `PlaneMesh` with just some different default settings, it's mostly a convenience as this is often used for billboarding. So I would use `PlaneMesh` in most scenarios.